### PR TITLE
treewide: sha512 → hash

### PR DESCRIPTION
**Pull request by nicoonoclaste**
## Description of changes

- Improve `sha-to-hash.py` to
  - handle multiple hash functions (currently SHA-256 and SHA-512),
  - automatically skip autogenerated files, based on heuristics.

- Replace `sha512` with `hash` through nixpkgs, where appropriate.
  This does not include autogenerated files; the tools making those should be updated & rerun.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/browsers/firefox/packages.nix
Comment (mweinelt) [inline pkgs/applications/networking/browsers/firefox/packages.nix:1]: These are in sha512 for the reason that the upstream provides these hashes in a signed file. Our updater verifies the gpg signature on the files, so we really want to keep them like this for the time being.

**Comment by vcunat** [comment]
For reference, lots of discussion around this happened at https://github.com/NixOS/rfcs/pull/131

**Comment by vcunat** [comment]
That sounds like it stems from a split in the community where to move on this topic (whether to prefer SRI and how strongly), which seems very much like that RFC.

**Comment by vcunat** [comment]
From what you say I'd think it would be better to document them.  Even if they would be considered deprecated for new packages or something, in which case you just express that in those docs (though I'm not aware of such a decision so far).

**Review by ulrikstrid — APPROVED** [review]
From the point of view of OCaml we have already been moving this way so this is fine. I have not reviewed the script(s).

File: maintainers/scripts/sha-to-SRI.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/env nix-shell
```
Comment (AndersonTorres) [inline maintainers/scripts/sha-to-SRI.py:1]: rename the file to `sha-to-sri.py`.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by AndersonTorres — APPROVED** [review]
I don't know about pretty printers in Python.
Besides, LGTM

File: pkgs/applications/networking/browsers/firefox/packages.nix
Comment (vcunat) [reply pkgs/applications/networking/browsers/firefox/packages.nix:1]: Oh, it wasn't said explicitly that `thunderbird` is the same?  Well, no matter anyway 🤷🏽

Comment (vcunat) [reply pkgs/applications/networking/browsers/firefox/packages.nix:1]: Oh, not by itself as it broke down the updater, but I can deal with it easily.

**Pull request merged**